### PR TITLE
Django 2.0 compatible

### PR DESCRIPTION
**Pull request by dewangmodi**
fixing issue#3

### CODE REVIEW

**Comment by mohitkh7** [comment]
Also update requirements.txt file

**Comment by dewangmodi** [comment]
done

**Comment by mohitkh7** [comment]
I am getting error related to AutoSlugField.
I doubt current version of Autoslugfield package we are using is not compatible with Django 2

**Comment by mohitkh7** [comment]
@dewangmodi This PR needs following changes - 
1. urls should be changed to path.
2. alternative for django-autoslug as it does not supports Django 2 [ref](https://github.com/justinmayer/django-autoslug/issues/50)
Making manual changes to library and changing `'django.core.urlresolvers` to django.urls` is an option, but it is not practical as every contributor will have to make this change.
We can try (this package)[https://pypi.org/project/django-autoslug-iplweb/#description]

**Comment by dewangmodi** [comment]
Please review the pull request again

File: learn/urls.py [hunk 1]
```diff
@@ -3,60 +3,95 @@
...
+    path('reset/<slug:uidb64>/<str:token>/',PasswordResetConfirmView.as_view, name='password_reset_confirm'),
+    #url(r'^reset/done/$', PasswordResetCompleteView.as_view(), name='password_reset_complete'),
+    path('reset/done/', PasswordResetCompleteView.as_view(), name='password_reset_complete'),
 ]
 #to avoid errors
 urlpatterns+=[
-	url(r'^(?P<slug>[-\w]+)/$',TopicDetails.as_view(),name="TopicDetails"),
+	#url(r'^(?P<slug>[-\w]+)/$',TopicDetails.as_view(),name="TopicDetails"),
+	path('<slug:slug>/',TopicDetails.as_view(),name="TopicDetails"),
 	url(r'^(?P<slug>[-\w]+)/test.$',test,name="test"),
```
Comment (mohitkh7) [inline learn/urls.py]: You haven't changed this URLpattern from URL to path.

File: learn/urls.py [hunk 2]
```diff
@@ -3,60 +3,95 @@
 from django.conf.urls.static import static
 # from .views import TopicList,home,TopicDetails,TopicCreate,TopicUpdate,TopicDelete,ResourceCreate,ResourceUpdate,ResourceDelete,ReviewCreate,ReviewUpdate,ReviewDelete,SignupView, myaccount, UserUpdate, AllActivityList
 from .views import *
-from django.contrib.auth.views import LoginView,LogoutView,PasswordResetView,password_reset_done,password_reset,password_reset_confirm,password_reset_complete,PasswordChangeView,PasswordChangeDoneView
+#from django.contrib.auth.views import LoginView,LogoutView,PasswordResetView,password_reset_done,password_reset,password_reset_confirm,password_reset_complete,PasswordChangeView,PasswordChangeDoneView
```
Comment (mohitkh7) [inline learn/urls.py]: Instead of removing you have commented all the previous code. 
Any specific reason for this ?

File: learn/urls.py [hunk 3]
```diff
@@ -3,60 +3,65 @@
...
-    url(r'^reset/done/$', password_reset_complete, name='password_reset_complete'),
+    path('reset/<slug:uidb64>/<str:token>/',PasswordResetConfirmView.as_view, name='password_reset_confirm'),
+    path('reset/done/', PasswordResetCompleteView.as_view(), name='password_reset_complete'),
 ]
 #to avoid errors
 urlpatterns+=[
-	url(r'^(?P<slug>[-\w]+)/$',TopicDetails.as_view(),name="TopicDetails"),
-	url(r'^(?P<slug>[-\w]+)/test.$',test,name="test"),
+	path('<slug:slug>/',TopicDetails.as_view(),name="TopicDetails"),
+	path('<slug:slug>/test.$',test,name="test"),
```
Comment (mohitkh7) [inline learn/urls.py]: There is an warning on this line 
`?: (2_0.W001) Your URL pattern '<slug:slug>/test.$' [name='test'] has a route that contains '(?P<', begins with a '^', or ends with a '$'. This was likely an oversight when migrating to django.urls.path().`
Remove `$` from this statement.

**Pull request merged**